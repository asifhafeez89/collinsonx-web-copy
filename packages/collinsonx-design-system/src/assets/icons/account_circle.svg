<svg  viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="account_circle">
<path id="Vector" d="M15.9998 2.6665C8.63984 2.6665 2.6665 8.63984 2.6665 15.9998C2.6665 23.3598 8.63984 29.3332 15.9998 29.3332C23.3598 29.3332 29.3332 23.3598 29.3332 15.9998C29.3332 8.63984 23.3598 2.6665 15.9998 2.6665ZM9.4265 24.3732C9.99984 23.1732 13.4932 21.9998 15.9998 21.9998C18.5065 21.9998 22.0132 23.1732 22.5732 24.3732C20.7598 25.8132 18.4798 26.6665 15.9998 26.6665C13.5198 26.6665 11.2398 25.8132 9.4265 24.3732ZM24.4798 22.4398C22.5732 20.1198 17.9465 19.3332 15.9998 19.3332C14.0532 19.3332 9.4265 20.1198 7.51984 22.4398C6.15984 20.6532 5.33317 18.4265 5.33317 15.9998C5.33317 10.1198 10.1198 5.33317 15.9998 5.33317C21.8798 5.33317 26.6665 10.1198 26.6665 15.9998C26.6665 18.4265 25.8398 20.6532 24.4798 22.4398ZM15.9998 7.99984C13.4132 7.99984 11.3332 10.0798 11.3332 12.6665C11.3332 15.2532 13.4132 17.3332 15.9998 17.3332C18.5865 17.3332 20.6665 15.2532 20.6665 12.6665C20.6665 10.0798 18.5865 7.99984 15.9998 7.99984ZM15.9998 14.6665C14.8932 14.6665 13.9998 13.7732 13.9998 12.6665C13.9998 11.5598 14.8932 10.6665 15.9998 10.6665C17.1065 10.6665 17.9998 11.5598 17.9998 12.6665C17.9998 13.7732 17.1065 14.6665 15.9998 14.6665Z" fill="#202020"/>
</g>
</svg>
