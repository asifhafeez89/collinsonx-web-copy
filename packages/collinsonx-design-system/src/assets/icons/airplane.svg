<svg width="14" height="12" viewBox="0 0 14 12" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.680234 10.4771H13.3469V11.8104H0.680234V10.4771ZM13.7269 4.2371C13.5869 3.70376 13.0336 3.38376 12.5002 3.53043L8.96023 4.4771L4.36023 0.19043L3.07357 0.53043L5.83357 5.31043L2.52023 6.1971L1.2069 5.17043L0.240234 5.43043L1.9669 8.42376L13.0136 5.47043C13.5536 5.3171 13.8669 4.77043 13.7269 4.2371Z" fill="#0C8599"/>
</svg>
