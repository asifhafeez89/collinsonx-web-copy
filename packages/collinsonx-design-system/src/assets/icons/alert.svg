<svg  viewBox="0 0 21 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M8.72313 1.85022L1.23668 14.3484C1.08244 14.6157 1.00084 14.9187 1.00001 15.2273C0.999174 15.5359 1.07914 15.8394 1.23194 16.1075C1.38474 16.3756 1.60507 16.5991 1.87101 16.7557C2.13695 16.9122 2.43925 16.9965 2.74784 17H17.7216C18.0302 16.9965 18.3325 16.9122 18.5984 16.7557C18.8644 16.5991 19.0847 16.3756 19.2375 16.1075C19.3903 15.8394 19.4703 15.5359 19.4694 15.2273C19.4686 14.9187 19.387 14.6157 19.2328 14.3484L11.7454 1.85022C11.5878 1.59065 11.3659 1.37606 11.1012 1.22715C10.8366 1.07823 10.538 1 10.2343 1C9.93059 1 9.63201 1.07823 9.36733 1.22715C9.10264 1.37606 8.88078 1.59065 8.72313 1.85022Z" stroke="#D42A54" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M10.2877 6.33302V9.88868" stroke="#D42A54" stroke-width="2.75" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M10.2877 13.4443H10.2966" stroke="#D42A54" stroke-width="2.75" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
