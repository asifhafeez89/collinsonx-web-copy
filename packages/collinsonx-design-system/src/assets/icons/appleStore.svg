<svg width="17" height="16" viewBox="0 0 17 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_4119_1132)">
<path fill-rule="evenodd" clip-rule="evenodd" d="M16.5 3.46667V12.5333C16.5009 12.9888 16.4119 13.44 16.238 13.8611C16.0641 14.2821 15.8088 14.6646 15.4867 14.9867C15.1646 15.3088 14.7821 15.5641 14.3611 15.738C13.94 15.9119 13.4888 16.0009 13.0333 16H3.96667C3.51117 16.0009 3.05997 15.9119 2.63896 15.738C2.21796 15.5641 1.83543 15.3088 1.51334 14.9867C1.19125 14.6646 0.935932 14.2821 0.762038 13.8611C0.588144 13.44 0.499097 12.9888 0.500007 12.5333V3.46667C0.499097 3.01117 0.588144 2.55997 0.762038 2.13896C0.935932 1.71796 1.19125 1.33543 1.51334 1.01334C1.83543 0.69125 2.21796 0.435932 2.63896 0.262038C3.05997 0.0881437 3.51117 -0.000903146 3.96667 6.90559e-06H13.0333C13.4888 -0.000903146 13.94 0.0881437 14.3611 0.262038C14.7821 0.435932 15.1646 0.69125 15.4867 1.01334C15.8088 1.33543 16.0641 1.71796 16.238 2.13896C16.4119 2.55997 16.5009 3.01117 16.5 3.46667Z" fill="url(#paint0_linear_4119_1132)"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M11.5134 9.19995H13.5001C13.6946 9.19995 13.8811 9.27721 14.0186 9.41474C14.1562 9.55227 14.2334 9.73879 14.2334 9.93328C14.2334 10.1278 14.1562 10.3143 14.0186 10.4518C13.8811 10.5894 13.6946 10.6666 13.5001 10.6666H11.5134V9.19995Z" fill="white"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M9.07327 9.19999C9.21183 9.19916 9.34917 9.22584 9.47734 9.27848C9.60551 9.33112 9.72196 9.40868 9.81993 9.50665C9.91791 9.60463 9.99547 9.72108 10.0481 9.84925C10.1007 9.97742 10.1274 10.1148 10.1266 10.2533C10.1271 10.395 10.1 10.5354 10.0466 10.6667H3.49993C3.30544 10.6667 3.11892 10.5894 2.98139 10.4519C2.84386 10.3143 2.7666 10.1278 2.7666 9.93332C2.7666 9.73883 2.84386 9.5523 2.98139 9.41478C3.11892 9.27725 3.30544 9.19999 3.49993 9.19999H9.07327ZM8.8866 4.37332V5.65332H8.03327L6.99327 3.86665C6.94759 3.78785 6.91789 3.70082 6.90587 3.61053C6.89386 3.52025 6.89976 3.42848 6.92324 3.34047C6.94673 3.25247 6.98733 3.16995 7.04273 3.09766C7.09813 3.02536 7.16723 2.96469 7.2461 2.91912C7.32497 2.87355 7.41204 2.84398 7.50235 2.8321C7.59265 2.82021 7.68441 2.82625 7.77238 2.84986C7.86035 2.87347 7.94281 2.91419 8.01503 2.96969C8.08725 3.02519 8.14781 3.09439 8.19327 3.17332L8.8866 4.37332ZM9.61993 5.61332L13.2599 11.92C13.3572 12.088 13.3837 12.2877 13.3337 12.4752C13.2837 12.6628 13.1612 12.8227 12.9933 12.92C12.8253 13.0172 12.6256 13.0438 12.438 12.9938C12.2505 12.9438 12.0905 12.8213 11.9933 12.6533L9.79327 8.82665C9.59327 8.45332 9.4466 8.15999 9.35327 7.99999C9.13415 7.62668 9.04172 7.1925 9.08979 6.76232C9.13785 6.33214 9.32383 5.92906 9.61993 5.61332Z" fill="white"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M6.63336 9.8667H4.94003L8.7667 3.24003C8.86394 3.07206 9.02393 2.9496 9.21147 2.89959C9.399 2.84958 9.59873 2.87612 9.7667 2.97336C9.93467 3.07061 10.0571 3.2306 10.1071 3.41813C10.1571 3.60567 10.1306 3.80539 10.0334 3.97336L6.63336 9.8667ZM5.74003 11.4L5.03336 12.64C4.98521 12.7232 4.92115 12.7961 4.84484 12.8545C4.76852 12.9129 4.68145 12.9557 4.58859 12.9805C4.49573 13.0052 4.39891 13.0115 4.30364 12.9988C4.20837 12.9861 4.11653 12.9548 4.03336 12.9067C3.95019 12.8585 3.87732 12.7945 3.81891 12.7182C3.76049 12.6419 3.71769 12.5548 3.69292 12.4619C3.66816 12.3691 3.66193 12.2722 3.67459 12.177C3.68725 12.0817 3.71854 11.9899 3.7667 11.9067L4.2867 11C4.44818 10.9197 4.62634 10.8786 4.8067 10.88C4.99481 10.8739 5.18106 10.919 5.34548 11.0106C5.5099 11.1022 5.64629 11.2368 5.74003 11.4Z" fill="white"/>
</g>
<defs>
<linearGradient id="paint0_linear_4119_1132" x1="8.49987" y1="16" x2="8.49987" y2="7.62939e-06" gradientUnits="userSpaceOnUse">
<stop stop-color="#1D6FF2"/>
<stop offset="1" stop-color="#1AC8FC"/>
</linearGradient>
<clipPath id="clip0_4119_1132">
<rect width="16" height="16" fill="white" transform="translate(0.5)"/>
</clipPath>
</defs>
</svg>
