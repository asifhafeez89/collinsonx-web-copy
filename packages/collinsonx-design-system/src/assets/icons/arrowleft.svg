<svg width="17" height="10" viewBox="0 0 17 10" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5.5 0.5L1 4.902L5.5 9.5" stroke="#827127"/>
<path d="M1.5 5H15.54" stroke="#827127" stroke-linecap="square"/>
</svg>
