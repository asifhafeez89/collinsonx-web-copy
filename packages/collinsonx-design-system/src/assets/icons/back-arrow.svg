<svg viewBox="0 0 20 37" fill="none" xmlns="http://www.w3.org/2000/svg">
<line x1="1" y1="20" x2="19" y2="20" stroke="#25262B" stroke-width="2" stroke-linecap="round"/>
<path d="M7 26L1 20L7 14" stroke="#25262B" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
