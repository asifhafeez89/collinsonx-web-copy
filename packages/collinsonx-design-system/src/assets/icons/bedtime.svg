<svg width="19" height="20" viewBox="0 0 19 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.43992 2.48488C5.80992 10.0249 11.1899 14.8949 15.0999 16.2849C13.7099 17.3749 11.9799 17.9949 10.1699 17.9949C5.75992 17.9949 2.16992 14.4049 2.16992 9.99488C2.16992 6.54488 4.36992 3.59488 7.43992 2.48488ZM10.1599 0.00488281C4.56992 0.00488281 0.169922 4.53488 0.169922 9.99488C0.169922 15.5149 4.64992 19.9949 10.1699 19.9949C13.8799 19.9949 17.0999 17.9749 18.8299 14.9749C11.3199 14.7249 6.73992 6.54488 10.5099 0.00488281C10.3899 0.00488281 10.2799 0.00488281 10.1599 0.00488281Z" fill="#202020"/>
</svg>
