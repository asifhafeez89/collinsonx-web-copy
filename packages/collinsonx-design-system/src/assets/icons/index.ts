export { default as Bell } from './bell.svg';
export { default as BurgerNav } from './burger-nav.svg';
export { default as Cart } from './cart.svg';
export { default as Chat } from './chat.svg';
export { default as Check } from './check.svg';
export { default as ChevronLeft } from './chevron-left.svg';
export { default as ChevronRight } from './chevron-right.svg';
export { default as Clock } from './clock.svg';
export { default as Close } from './close.svg';
export { default as ExperienceX } from './experience-x.svg';
export { default as Filter } from './filter.svg';
export { default as Home } from './home.svg';
export { default as Lounge } from './lounge.svg';
export { default as MapPin } from './map-pin.svg';
export { default as Search } from './search.svg';
export { default as Coffee } from './coffee.svg';
export { default as Calendar } from './calendar.svg';
export { default as Logout } from './logout.svg';
export { default as Pending } from './pending.svg';
export { default as Confirmed } from './confirmed.svg';
export { default as Declined } from './declined.svg';
export { default as BackArrow } from './back-arrow.svg';
export { default as Warning } from './warning.svg';
export { default as Edit } from './edit.svg';
export { default as Aeroplane } from './airplane.svg';
export { default as CalendarB } from './calendarblue.svg';
export { default as Magglass } from './magglass.svg';
