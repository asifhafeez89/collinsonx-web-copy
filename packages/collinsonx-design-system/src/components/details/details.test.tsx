import renderer from 'react-test-renderer';
import Details from '.';
import { MapPin, Clock } from '../../assets/icons';

describe('<Details />', () => {
  it('renders Details', () => {
    const infos = [
      {
        header: 'Heathrow',
        description: 'Terminal 5',
        icon: <MapPin width={16} color="#0C8599" />,
      },
      {
        header: 'Mon - Sun | 05:00 - 22:00',
        description: 'Walk-in available',
        icon: <Clock width={16} color="#0C8599" />,
      },
    ];

    const tree = renderer
      .create(<Details title="This is a title" infos={infos} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
