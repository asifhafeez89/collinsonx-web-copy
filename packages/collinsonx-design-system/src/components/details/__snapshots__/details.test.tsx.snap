// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders Details 1`] = `
.emotion-0 {
  background: #fff;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 0.5rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-weight: 600;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1 emotion-2"
  >
    <div
      className="emotion-3"
    >
      <div
        className="emotion-4"
      >
        <test-file-stub
          color="#0C8599"
          width={16}
        />
      </div>
      <div
        className="emotion-5"
      >
        Heathrow
      </div>
      <div
        className="emotion-6"
      >
        Terminal 5
      </div>
    </div>
    <div
      className="emotion-3"
    >
      <div
        className="emotion-4"
      >
        <test-file-stub
          color="#0C8599"
          width={16}
        />
      </div>
      <div
        className="emotion-5"
      >
        Mon - Sun | 05:00 - 22:00
      </div>
      <div
        className="emotion-6"
      >
        Walk-in available
      </div>
    </div>
  </div>
</div>
`;
